src/utils.js

```diff
@@ -110,10 +110,9 @@ export function getXAttrs(el, component, type) {
         directives = directives.concat(Object.entries(spreadObject).map(([name, value]) => parseHtmlAttribute({ name, value })))
     }
     
-    // If no type is passed in for filtering, bypass filter
-    if (!type) return directives;
+    if (type) return directives.filter(i => i.type === type)
 
-    return directives.filter(i => i.type === type)
+    return directives;
```

Comment:
After a moment realized that `type` parameter of the function [getXAttrs](https://github.com/alpinejs/alpine/pull/607/commits/f761f02bccf476f6f13d7a459dee8c704aa627c8#diff-2b4ca49d4bb0a774c4d4c1672d7aa781R100) is optional accordingly to these use-cases in `components.js` file:
1. https://github.com/alpinejs/alpine/blob/a0fab06379b5427df82a8a32e495e373f51c7a16/src/component.js#L165
2. https://github.com/alpinejs/alpine/blob/a0fab06379b5427df82a8a32e495e373f51c7a16/src/component.js#L227
3. https://github.com/alpinejs/alpine/blob/a0fab06379b5427df82a8a32e495e373f51c7a16/src/component.js#L243

Our case is: `type` param is optional and happy/main case is to just get `x-*` attributes.
In case the `type` parameter was required then the happy/main case would be the filtering-by-type.

So it's better to return optional case wrapped in optional condition itself and return happy/main case by default.